### PR TITLE
Turn anitya cronjob to service

### LINKED ISSUE #668: Turn anitya cron job to service
Current cron job used by Anitya is not very reliable.

So it will be better to turn it into service probably using [twisted](https://twistedmatrix.com/trac/) library. This will make scheduling much easier and we could also add the ability to send e-mail when things are failing.

Here is the list of the features it should have:
- easy scheduling
- notification about high fail rate (probably through e-mail)

### PR DESCRIPTION
**Pull request by Zlopez**
Introduce check_service to Anitya. This service will check for new
versions of monitored projects in loop with 5 minutes wait time.
It will use the already existing field next_check in project to know
which project is ready to check. Projects that were blocked by ratelimit
will be prioritised.
At the end of each run it will add some statistics to db (success, failure,
blocked).

Fixes #668 

Signed-off-by: Michal Konečný <mkonecny@redhat.com>

### CODE REVIEW

**Comment by Zlopez** [comment]
This pull request **fixes 2 alerts** when merging 122cfdb37593df1a7a2950aeb0ec849a1f44ce8a into cf3ad42e7a57ee248fe1877e4e0f0514298cbf15 - [view on LGTM.com](https://lgtm.com/projects/g/release-monitoring/anitya/rev/pr-5eba6eb6c50ca7abb5a8532c3b104f122dab5956)

**fixed alerts:**

* 1 for Module is imported with &#39;import&#39; and &#39;import from&#39;
* 1 for Nested loops with same variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: anitya/check_service.py [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+from time import sleep
+
+# We need to use multiprocessing.dummy, since we use the Pool to run
+# update_project. This in turn uses anitya.lib.backends.BaseBackend.call_url,
+# which utilizes a global requests session. Requests session is not usable
+# with multiprocessing, since it would need to share the SSL connection between
+# processes (see https://stackoverflow.com/q/3724900#3724938).
+# multiprocessing.dummy.Pool is in fact a Thread pool, which works ok
+# with a global shared requests session.
+import multiprocessing.dummy as multiprocessing
```
Comment (cverna) [inline anitya/check_service.py]: You could also use `concurrent.futures.ThreadPoolExecutor` that is available in Python3 

https://docs.python.org/3/library/concurrent.futures.html#module-concurrent.futures

File: setup.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def get_requirements(requirements_file="requirements.txt"):
     ],
     packages=find_packages(exclude=["anitya.tests", "anitya.tests.*"]),
     include_package_data=True,
-    scripts=["files/anitya_cron.py", "anitya/sar.py"],
+    scripts=["anitya/check_service.py", "anitya/sar.py"],
```
Comment (cverna) [inline setup.py]: This service will run in a new pod in Openshift ?

Comment (Zlopez) [reply setup.py]: Yes, instead of Cronjob I will have new pod running.

File: anitya/check_service.py (hunk 1 above)
Comment (Zlopez) [reply anitya/check_service.py]: Thanks, I will look into it. I used what was originally in cron job.

File: setup.py (hunk 2 above)
Comment (cverna) [reply setup.py]: Ok cool :+1:

**Comment by Zlopez** [comment]
This pull request **fixes 2 alerts** when merging 530989995a60fda661e3b3fc1f3b551bfe54cc9b into 7634681700c46e3c7575d924799238b7f5f681c4 - [view on LGTM.com](https://lgtm.com/projects/g/release-monitoring/anitya/rev/pr-7deb2b8ccc3a005ca8f64063be6538fce6f9c810)

**fixed alerts:**

* 1 for Module is imported with &#39;import&#39; and &#39;import from&#39;
* 1 for Nested loops with same variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by Zlopez** [comment]
This pull request **fixes 2 alerts** when merging 2cc6fc07120f1953ca8feafed5567f7e5484a65e into 619c47fb470cb5db4889a91585cae7b0c4d11c99 - [view on LGTM.com](https://lgtm.com/projects/g/release-monitoring/anitya/rev/pr-2c2316f1e8dfc389b1ba1c2d8c27193883b5e428)

**fixed alerts:**

* 1 for Module is imported with &#39;import&#39; and &#39;import from&#39;
* 1 for Nested loops with same variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**